data-plane/config/broker/100-config-kafka-broker-data-plane.yaml

```diff
@@ -133,7 +133,7 @@ data:
     isolation.level=read_uncommitted
     max.poll.interval.ms=300000
     max.poll.records=500
-    # partition.assignment.strategy=org.apache.kafka.clients.consumer.CooperativeStickyAssignor
+    partition.assignment.strategy=org.apache.kafka.clients.consumer.CooperativeStickyAssignor
```

Comment:
The problem with this is that it's supported only on Kafka 2.4+, so shipping this config by default isn't optimal since there are lots of people on old Kafka clusters.

I think we can get the same benefits by using the StickyAssignor with a smarter logic that records the last assignment as suggested in the docs here https://kafka.apache.org/26/javadoc/org/apache/kafka/clients/consumer/StickyAssignor.html:

```
 class TheNewRebalanceListener implements ConsumerRebalanceListener {
   Collection<TopicPartition> lastAssignment = Collections.emptyList();

   void onPartitionsRevoked(Collection<TopicPartition> partitions) {
     for (TopicPartition partition: partitions)
       commitOffsets(partition);
   }

   void onPartitionsAssigned(Collection<TopicPartition> assignment) {
     for (TopicPartition partition: difference(lastAssignment, assignment))
       cleanupState(partition);

     for (TopicPartition partition: difference(assignment, lastAssignment))
       initializeState(partition);

     for (TopicPartition partition: assignment)
       initializeOffset(partition);

     this.lastAssignment = assignment;
   }
 }
```

what do you think?

Reply:
> The problem with this is that it's supported only on Kafka 2.4+,

what version of Kafka do we generally support? (as in a range)

Reply:
@pierDipi @matzew  I think we'll need to experiment with this logic a bit to see if it gives the same low number of duplicate results as `CooperativeStickyAssignor` especially since this sticky+smart logic may be still be a semi-eager protocol. What do you think about changing the current `Range` default to `Sticky` anyways for release `1.6.0`? And then customizing it a bit more after we've done some testing with results..

IBM Event Streams is on Kafka v2.8 so they should probably use Cooperative...